ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -586,4 +670,22 @@ private static ByteRange buildByteRange(String rangeHeaderValue) throws RestServ
     }
     return range;
   }
+
+  /**
+   * Sets an argument into the {@link RestRequest}.
+   * @param restRequest The {@link RestRequest} to set the argument. Cannot be {@code null}.
+   * @param key The key of the argument. Cannot be {@code null}.
+   * @param value The value of the argument.
+   * @param isOldValueAllowed {@code true} if an old argument value is allowed; {@code false} otherwise.
+   * @return The old value of the argument.
+   */
+  public static Object setArg(RestRequest restRequest, String key, Object value, boolean isOldValueAllowed) {
```

Comment:
not sure this function is required. The `requireNonNull` statements look superfluous.